host/sgx/loadelf.c

```diff
@@ -1206,43 +1237,24 @@ static oe_result_t _patch_elf_image(
     oeprops->image_info.heap_rva =
         oeprops->image_info.reloc_rva + oeprops->image_info.reloc_size;
 
-    /* TODO: Remove these as special globals once thread init for multiple
-     * modules is added */
-    if (image->link_info.tdata_size)
-    {
-        _set_uint64_t_dynamic_symbol_value(
-            image, "_tdata_rva", image->link_info.tdata_rva);
-        _set_uint64_t_dynamic_symbol_value(
-            image, "_tdata_size", image->link_info.tdata_size);
-        _set_uint64_t_dynamic_symbol_value(
-            image, "_tdata_align", image->link_info.tdata_align);
-
-        aligned_size += oe_round_up_to_multiple(
-            image->link_info.tdata_size, image->link_info.tdata_align);
-    }
-    if (image->link_info.tbss_size)
-    {
-        _set_uint64_t_dynamic_symbol_value(
-            image, "_tbss_size", image->link_info.tbss_size);
-        _set_uint64_t_dynamic_symbol_value(
-            image, "_tbss_align", image->link_info.tbss_align);
-
-        aligned_size += oe_round_up_to_multiple(
-            image->link_info.tbss_size, image->link_info.tbss_align);
-    }
-
-    aligned_size = oe_round_up_to_multiple(aligned_size, OE_PAGE_SIZE);
-    _set_uint64_t_dynamic_symbol_value(
-        image,
-        "_td_from_tcs_offset",
-        aligned_size + OE_SGX_NUM_CONTROL_PAGES * OE_PAGE_SIZE);
-    context->num_tls_pages = aligned_size / OE_PAGE_SIZE;
-
     /* Clear the hash when taking the measure */
     memset(oeprops->sigstruct, 0, sizeof(oeprops->sigstruct));
 
     /* Dynamically link the enclave and its needed modules */
-    OE_CHECK(_link_elf_images(image, &module_base));
+    {
+        int64_t tls_start_offset = 0;
+        OE_CHECK(_link_elf_images(image, &module_base, &tls_start_offset));
+
+        // Compute the offset of td from total thread local size.
+        uint64_t total_tls_size = (uint64_t)-tls_start_offset;
```

Comment:
We should probably check that `tls_start_offset <= 0` for this right?

Reply:
Done.